lib/src/compiler/parse.rs

```diff
@@ -1,14 +1,220 @@
-use crate::{ast::Program, compiler::Compiler};
+use crate::{
+    ast::{Instr, LangValue, Program, StackIdentifier},
+    compiler::Compiler,
+    error::ParseError,
+};
 use failure::Fallible;
+use nom::{
+    branch::alt,
+    bytes::complete::tag,
+    character::complete::{char, digit1, multispace0},
+    combinator::map_res,
+    multi::{many0, many1},
+    sequence::{delimited, preceded, terminated},
+    IResult,
+};
 use std::io::Read;
 
+fn parse_read(input: &str) -> IResult<&str, Instr> {
+    let (input, _) = tag("Read")(input)?;
```

Comment:
What is the `_` value that you're throwing away in every function?

Reply:
most of the builtin funcs return a touple of (str,str). The first element is whats left to parse, the second is the string of what matched. So i just through away the match (since it was Read) in this case then return the actual instruction. Ill add a comment